data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ContainsSubstringExpressionFunctionTest.java

```diff
@@ -0,0 +1,86 @@
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.function.Function;
+
+class ContainsSubstringExpressionFunctionTest {
+    private ContainsSubstringExpressionFunction containsSubstringExpressionFunction;
+    private Event testEvent;
+    private Function<Object, Object> testFunction;
+    private List<Object> tags;
+    private String testKey;
+    private String testKey2;
+    private String testKey3;
+    private String testValue;
+    private String testValue2;
+
+    private Event createTestEvent(final Object data) {
+        return JacksonEvent.builder().withEventType("event").withData(data).build();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        testKey = RandomStringUtils.randomAlphabetic(5);
+        testKey2 = RandomStringUtils.randomAlphabetic(5);
+        testKey3 = RandomStringUtils.randomAlphabetic(5);
+        testValue = RandomStringUtils.randomAlphabetic(10);
+        testValue2 = testValue.substring(2,6);
+        testEvent = createTestEvent(Map.of(testKey, testValue, testKey2, testValue2, testKey3, 1234));
+        tags = new ArrayList<>();
+        testFunction = mock(Function.class);
+    }
+
+    public ContainsSubstringExpressionFunction createObjectUnderTest() {
+        return new ContainsSubstringExpressionFunction();
+    }
+
+    @Test
+    void testContainsSubstringBasic() {
+        containsSubstringExpressionFunction = createObjectUnderTest();
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("\"abcde\"", "\"abcd\""), testEvent, testFunction), equalTo(true));
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("/"+testKey, "/"+testKey2), testEvent, testFunction), equalTo(true));
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("\""+testValue+"\"", "/"+testKey2), testEvent, testFunction), equalTo(true));
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("/"+testKey, "\""+testValue2+"\""), testEvent, testFunction), equalTo(true));
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("\"abcde\"", "\"xyz\""), testEvent, testFunction), equalTo(false));
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("/unknown", "xyz"), testEvent, testFunction), equalTo(false));
+        assertThat(containsSubstringExpressionFunction.evaluate(List.of("\"abcde\"", "/unknown"), testEvent, testFunction), equalTo(false));
+    }
+
+    @Test
+    void testContainsSubstringMultipleSubstrings() {
+        containsSubstringExpressionFunction = createObjectUnderTest();
+        String testString = RandomStringUtils.randomAlphabetic(10);
+        for (int i = 1; i < testString.length() - 1; i++) {
+            assertThat(containsSubstringExpressionFunction.evaluate(List.of("\""+testString+"\"", "\""+testString.substring(0, i)+"\""), testEvent, testFunction), equalTo(true));
+        }
+    }
+
+    @Test
+    void testInvalidContainsSubstring() {
```

Comment:
Take it or leave it: These tests may be simplified by using `@ParameterizedTest`. That would require refactoring to ensure the event is created statically as well. This would reduce the number assertions and allow anyone to extend the test by focusing just on invalid values instead of writing the repeated line.

Reply:
I don't think that would work because the type is a list. Which means I have to write another `methodSource` that generates a list because `valueSource` does not take a list. If I end up writing a method, then I don't think there is much difference between the two tests.